Maple2.Model/Metadata/AchievementMetadata.cs

```diff
@@ -0,0 +1,37 @@
+﻿using System.Collections.Generic;
+using System.Numerics;
+using Maple2.Model.Enum;
+
+namespace Maple2.Model.Metadata;
+
+public record AchievementMetadata(
+    int Id,
+    string? Name,
+    bool AccountWide,
+    AchievementCategory Category,
+    string[] CategoryTags,
+    IReadOnlyDictionary<int, AchievementMetadataGrade> Grades) : ISearchResult;
+
+public record AchievementMetadataGrade(
+    AchievementMetadataCondition Condition,
+    AchievementMetadataReward? Reward);
+
+public record AchievementMetadataCondition(
+    AchievementConditionType Type,
+    AchievementMetadataCondition.Parameters? Codes,
+    long Value,
+    AchievementMetadataCondition.Parameters? Target) {
+
+    public record Parameters(
```

Comment:
what about setting all of these with default values of null so you don't have to pass null every time. (you only set 1 of the values anyways)

```cs
public record Paramters(string[]? Strings = null, Range<int>? Range = null, int[]? Integers = null);
```